src/prng.rs

```diff
@@ -3,42 +3,95 @@
 
 use super::field::{FieldElement, FieldError};
 use aes::{
-    cipher::{generic_array::GenericArray, FromBlockCipher, NewBlockCipher, StreamCipher},
+    cipher::{
+        generic_array::GenericArray, FromBlockCipher, NewBlockCipher,
+        StreamCipher as AesStreamCipher,
+    },
     Aes128, Aes128Ctr,
 };
+use blake3;
 use getrandom::getrandom;
 
 use std::marker::PhantomData;
 
-const BLOCK_SIZE: usize = 16;
 const DEFAULT_BUFFER_SIZE_IN_ELEMENTS: usize = 128;
 const MAXIMUM_BUFFER_SIZE_IN_ELEMENTS: usize = 4096;
-pub const SEED_LENGTH: usize = 2 * BLOCK_SIZE;
 
 pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, getrandom::Error> {
-    let mut seed = vec![0; SEED_LENGTH];
-    getrandom(seed.as_mut_slice())?;
+    let mut seed = [0; STREAM_CIPHER_AES128CTR_KEY_LENGTH];
+    getrandom(&mut seed)?;
 
     // get prng array
-    let data: Vec<F> = Prng::new_with_seed_and_optional_length(&seed, Some(share1.len())).collect();
+    let data: Vec<F> =
+        Prng::<_, Aes128Ctr, STREAM_CIPHER_AES128CTR_KEY_LENGTH>::new_with_seed_and_optional_length(
+            seed,
+            Some(share1.len()),
+        )
+        .collect();
 
     // secret share
     for (s1, d) in share1.iter_mut().zip(data.iter()) {
         *s1 -= *d;
     }
 
-    Ok(seed)
+    Ok(seed.to_vec())
 }
 
 pub(crate) fn extract_share_from_seed<F: FieldElement>(
     length: usize,
     seed: &[u8],
 ) -> Result<Vec<F>, PrngError> {
-    if seed.len() != 2 * BLOCK_SIZE {
+    if seed.len() != STREAM_CIPHER_AES128CTR_KEY_LENGTH {
         return Err(PrngError::SeedLen);
     }
 
-    Ok(Prng::new_with_seed_and_optional_length(seed, Some(length)).collect())
+    let mut seed_copy = [0; STREAM_CIPHER_AES128CTR_KEY_LENGTH];
+    seed_copy.clone_from_slice(seed);
+    Ok(
+        Prng::<_, Aes128Ctr, STREAM_CIPHER_AES128CTR_KEY_LENGTH>::new_with_seed_and_optional_length(
+            seed_copy,
+            Some(length),
+        )
+        .collect(),
+    )
+}
+
+/// A stream cipher maps a short key to a stream of pseudorandom bytes.
+pub(crate) trait StreamCipher<const KEY_LENGTH: usize> {
+    /// Returns a new stream ciphered keyed by `key`.
+    fn new(key: [u8; KEY_LENGTH]) -> Self;
```

Comment:
Instead of a generic const, ideally `KEY_LENGTH` would be a trait parameter. However, we can't do
```rust
fn new(key: [u8; Self::KEY_LENGTH]) -> Self;
```